src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php

```diff
@@ -131,16 +133,17 @@ public function unserialize($serialized)
     {
         $data = unserialize($serialized);
 
-        $this->name   = $data['name'];
-        $this->values = $data['values'];
+        $this->name        = $data['name'];
+        $this->values      = $data['values'];
+        $this->initialized = true;
```

Comment:
Yep indeed. But we don't have any way to serialize callables. So when we unserialize a dictionary not hydrated at first, it can't be hydrated anymore. This variable name is maybe misleading, because it's more to force to not hydrate after unserialization than anything else.

Btw, while talking with @PedroTroller about this fix he suggested to totally remove `\Serializable` from Dictionary interface. Now I've made my PR, I would agree with him: it's unnecessary, it allows to use dictionaries in unsupported fashions, and it's not really an extension point since the profiler template is tightly-coupled to this data structure.